engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -107,21 +133,34 @@ public void clear() {
 
     @Override
     public EntityBuilder newBuilder() {
-        return globalPool.newBuilder();
+        return getCurrentWorldPool().newBuilder();
     }
 
     @Override
     public EntityBuilder newBuilder(String prefabName) {
-        return globalPool.newBuilder(prefabName);
+        return getCurrentWorldPool().newBuilder(prefabName);
     }
 
     @Override
     public EntityBuilder newBuilder(Prefab prefab) {
-        return globalPool.newBuilder(prefab);
+        return getCurrentWorldPool().newBuilder(prefab);
     }
 
     @Override
     public EntityRef create() {
+        return getCurrentWorldPool().create();
+    }
+
+    @Override
+    public EntityRef create(GameManifest gameManifest) {
+        Map<String, WorldInfo> worldInfoMap = gameManifest.getWorldInfoMap();
+        worldManager = new WorldManager(gameManifest.getWorldInfo(TerasologyConstants.MAIN_WORLD));
+        for (Map.Entry<String, WorldInfo> worldInfoEntry : worldInfoMap.entrySet()) {
+            EngineEntityPool pool = new PojoEntityPool(this);
+            pool.create();
+            pools.add(pool);
+            worldManager.addWorldPool(worldInfoEntry.getValue(), pool);
+        }
         return globalPool.create();
     }
```

Comment:
returning the worldEntity and creating worldPools are different tasks, handle them differently.
Ideally, the worldEntity should come from getCurrentWorldPool().create(); (which should be the world pool for the world even in the case of a single world after the decoupling is complete)